libs/settings/include/settings/setting_collection.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include <vector>
+
+namespace fetch {
+namespace settings {
+
+class SettingBase;
+
+/**
+ * A very simple collection of pointers to settings object. The use must make sure that the
+ * lifetime of the settings is longer than this container
+ */
+class SettingCollection
+{
+public:
+  using Settings = std::vector<SettingBase *>;
```

Comment:
may this be a vector of unique_ptrs? Lifetime concerns would go away if so.

Reply:
Specifically the design doesn't want to control the lifetime of these objects. The `unique_ptr` would change this quite drastically. The closest version compatible version would be to use the `shared_ptr` or to use `std::reference_wrapper`. I did initially look at the reference wrapper and decided that it was more effort than was required.

The basic design here is that everything is designed to be very static so having this pointer vector is fine IMHO.

Reply:
ok